File: server/loans/core/src/main/java/ee/rsx/kata/bank/loans/domain/segment/gateway/FindCreditSegment.java
package ee.rsx.kata.bank.loans.domain.segment.gateway;

import ee.rsx.kata.bank.loans.domain.segment.CreditSegment;
import ee.rsx.kata.bank.loans.domain.ssn.SocialSecurityNumber;

import java.util.Optional;

@FunctionalInterface
public interface FindCreditSegment {

  Optional<CreditSegment> forPerson(SocialSecurityNumber ssn);
}

File: server/loans/core/src/main/kotlin/ee/rsx/kata/bank/loans/domain/segment/gateway/FindCreditSegment.kt
package ee.rsx.kata.bank.loans.domain.segment.gateway

import ee.rsx.kata.bank.loans.domain.segment.CreditSegment
import ee.rsx.kata.bank.loans.domain.ssn.SocialSecurityNumber

fun interface FindCreditSegment {
  operator fun invoke(forPerson: SocialSecurityNumber): CreditSegment?
}
